macros/clusterAnaScurve.py

```diff
@@ -0,0 +1,234 @@
+        pass
+    parsedTuple = parseListOfScanDatesFile(listOfScanDatesFile, alphaLabels=True)
+    listChamberAndScanDate = parsedTuple[0]
+
+    # Setup output scandates list
+    outputScanDatesName = listOfScanDatesFile.strip('.txt')
+    outputScanDatesName += "_Input4GemPlotter.txt"
+    outputScanDatesFile = open(outputScanDatesName, 'w+')
+    outputScanDatesFile.write('ChamberName\tscandate\n')
+
+    # invert chamber_config
+    from mapping.chamberInfo import chamber_config, GEBtype
+    linkByChamber = { value:key for key,value in chamber_config.iteritems() }
+    
+    # Make and launch a job for each file
+    import time
+    for idx,chamberAndScanDatePair in enumerate(listChamberAndScanDate):
+        # Setup the path
+        dirPath = getDirByAnaType(options.anaType, chamberAndScanDatePair[0], options.ztrim)
+        dirPath = "%s/%s"%(dirPath,chamberAndScanDatePair[1])
+
+        # Check if file exists, if it does not write to output as commented line but skip to next input
+        if not filePathExists(dirPath, tree_names[options.anaType][0]):
+            outputScanDatesFile.write('#%s\t%s\n'%(chamberAndScanDatePair[0],chamberAndScanDatePair[1]))
+            continue
+        outputScanDatesFile.write('%s\t%s\n'%(chamberAndScanDatePair[0],chamberAndScanDatePair[1]))
+
+        # Input file
+        jobInputFile = "%s/%s"%(dirPath, tree_names[options.anaType][0])
+        
+        # stdout
+        jobStdOut = "%s/stdout"%dirPath
+        runCommand( ["mkdir","-p", jobStdOut ] )
+        if len(os.listdir(jobStdOut)) > 0:
+            runCommand( ['rm','%s/jobOut.txt'%(jobStdOut) ] )
+            pass
+
+        # stderr
+        jobStdErr = "%s/stderr"%dirPath
+        runCommand( ["mkdir","-p", jobStdErr ] )
+        if len(os.listdir(jobStdErr)) > 0:
+            runCommand( ['rm','%s/jobErr.txt'%(jobStdErr) ] )
+            pass
+
+        # script to be run by the cluster
+        jobScriptName = "%s/clusterJob.sh"%dirPath
+        jobScript = open(jobScriptName, 'w+')
+        #jobScript.write('scl enable python27 - << \EOF\n')
+        jobScript.write('#!/bin/zsh\n')
+        jobScript.write('echo `python --version`\n')
+        jobScript.write('echo `gcc --version | grep "gcc"`\n')
+        jobScript.write('source /opt/rh/python27/enable\n')
+        jobScript.write('source /afs/cern.ch/sw/lcg/contrib/gcc/4.8.4/x86_64-slc6/setup.sh\n')
+        jobScript.write('export BUILD_HOME=%s\n'%buildHome)
+        jobScript.write('export DATA_PATH=%s\n'%dataPath)
+        jobScript.write('export ELOG_PATH=%s\n'%elogPath)
+        jobScript.write('source %s/cmsgemos/setup/paths.sh\n'%buildHome)
+        jobScript.write('source %s/gem-plotting-tools/setup/paths.sh\n'%buildHome)
+        jobScript.write('source $BUILD_HOME/venv/slc6/py27/bin/activate\n')
+        jobScript.write('echo `python --version`\n')
```

Comment:
This and next line are copy of lines 143-144. Do we need them in both places?

Reply:
They were exceptionally useful in debugging as on the worker node the python and gcc versions were not being set correctly.  If you object then I would prefer to remove the lines at 143-144 and keep 152-153 (as they reflect the version of python and gcc that the job is executed with)